perses/annihilation/relative.py

```diff
@@ -370,13 +370,13 @@ def _get_core_atoms(self):
                                                                      self._new_to_hybrid_map, name_of_residue)
 
         #The union of the two will give the core atoms that can result from either new or old topology
-        # TODO: Shouldn't these sets be the same?
         total_core_atoms = core_atoms_from_old.union(core_atoms_from_new)
+        assert sorted(core_atoms_from_old) == sorted(core_atoms_from_new), 'Core atoms must match between old and new systems'
```

Comment:
Should this use a `set()` comparison instead?

Reply:
I'll change this